tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc

```diff
@@ -96,8 +96,9 @@ Status SetTrtEngineInputs(nvinfer1::ICudaEngine* cuda_engine,
                           nvinfer1::IExecutionContext* execution_context,
                           const int trt_profile_idx,
                           std::vector<void*>& buffers, bool use_implicit_batch,
-                          int num_batch, OpKernelContext* ctx,
-                          const DataVec* input_vec) {
+                          int num_batch,
+                          const std::vector<std::vector<int32>>& shape_values,
+                          OpKernelContext* ctx, const DataVec* input_vec) {
```

Comment:
This is possible, but we need actually two inputs: the vector of Tensor* and a vector of input names. I had such an implementation earlier, which I rejected to keep `TRTEngineOp::ExecuteTrtEngine` shorter. I can bring this back.

Reply:
I was thinking about the proposed change. I am not convinced if it simplifies things. Let me know if I should go forward with this:

1. For the inputs it could work the following way: 
- we need a helper to construct the vector of tensors from `OpKernelContext` (since the `params_` field is private),
- another helper to map the unit test `DataVec -> gtl::InlinedVector<TensorValue,4>`.

2. The outputs need either to be allocated using `OpKernelContext`, or just reshaped in case of the unit tests. One would need two more helpers that return the vector of tensors: 
- create a helper functions doing the allocations for `OpKernelContext`
- another one for the reshape operations during unit test with `DataVec`.

My preferred solution would be to make the converter tests a subclass of OpTestBase, and run the tests as a TF op test. But that would be outside the scope of this PR.

Reply:
I think making the converter tests run as OpTestBase is a good idea.